### PR TITLE
Fix group select autocomplete

### LINKED ISSUE #13105: Enable app for group doesn't autocomplete anymore
* try to enable app for specific group only by clicking the checkbox on the app management page for text editor for example
* focus the input
* spinner appears and nothing is displayed even if I typed in the full group name

cc @DeepDiver1975 @jancborchardt

### PR DESCRIPTION
**Pull request by icewind1991**
Update js to reflect the changes in the returned json

Fixes #13105

cc @DeepDiver1975 @MorrisJobke

### CODE REVIEW

File: settings/js/settings.js [hunk 1]
```diff
@@ -45,25 +45,22 @@ OC.Settings = _.extend(OC.Settings, {
...
-									self._cachedGroups = results;
-								}
-								query.callback({results: results});
-							} else {
-								//FIXME add error handling
+							if (query.term === '') {
+								// cache full list
+								self._cachedGroups = results;
 							}
+							console.log(results);
```
Comment (MorrisJobke) [inline settings/js/settings.js]: Is this intended? I would like to drop this line

Comment (icewind1991) [inline settings/js/settings.js]: oops, will remove

**Comment by karlitschek** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Works :+1: except my comment above

**Comment by MorrisJobke** [comment]
JS unit tests run fine locally

**Pull request merged**